Sources/MapboxDirections/DirectionsOptions.swift

```diff
@@ -129,6 +129,8 @@ open class DirectionsOptions: Codable {
         self.waypoints = waypoints
         self.profileIdentifier = profileIdentifier ?? .automobile
         
+        if waypoints.count < 2 { assertionFailure("waypoints array requires at least 2 waypoints") }
```

Comment:
The `waypoints` property is read-write, so it’s valid and a common pattern to initialize the object first before filling in the waypoints. Instead of asserting here, move the assertion to where we build a request URL in `abridgedPath` or `Directions.url(forCalculating:httpMethod:)`.

https://github.com/mapbox/mapbox-directions-swift/blob/62b9689daa60fe0d9ee245868cb2bdaec3e09c95/Sources/MapboxDirections/DirectionsOptions.swift#L312